script/train-events.lua

```diff
@@ -121,6 +122,12 @@ function TrainArrives(train)
 
         setLamp(stop, "blue", 1)
 
+        end
+        -- if train had a delivery fire completed event
+        log("Train arrives at depot: "..train.id.." : "..serpent.block(delivery))
+        if delivery then
+          script.raise_event(on_delivery_completed_event, {train_id = train.id, shipment = delivery.shipment})

```

Comment:
on_delivery_completed_event is used when deliveries complete as intended
This state should be handled like delivery timeout and raise on_delivery_failed